packetbeat/protos/dhcpv4/option_ip_addresses.go

```diff
@@ -0,0 +1,85 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package dhcpv4
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+)
+
+// IPAddressesOption represents an option whose value is a list of IP addresses.
+type IPAddressesOption struct {
+	dhcpv4.OptionCode
+	IPAddresses []net.IP
+}
+
+// ParseIPAddressesOption returns a new IPAddressesOption from a byte
+// stream, or error if any.
+func ParseIPAddressesOption(data []byte) (*IPAddressesOption, error) {
+	if len(data) < 2 {
+		return nil, dhcpv4.ErrShortByteStream
+	}
+	code := dhcpv4.OptionCode(data[0])
+	length := int(data[1])
+	if length == 0 || length%4 != 0 {
+		return nil, fmt.Errorf("Invalid length: expected multiple of 4 larger than 4, got %v", length)
+	}
+	if len(data) < 2+length {
+		return nil, dhcpv4.ErrShortByteStream
+	}
+	servers := make([]net.IP, 0, length%4)
```

Comment:
nit: I think you want `length / 4` here

Reply:
Definitely. Good catch. I think I was looking at one of the parsers inside of the dhcp lib when I "wrote" that.

Will fix.

Reply:
I fixed that line and squashed the commit.